core/src/main/java/contacts/core/BroadQuery.kt

```diff
@@ -444,29 +444,23 @@ interface BroadQuery : Redactable {
     // I know that this interface also exist in Query but I want each API to have its own
     // interface for the results in case we need to deviate implementation. Besides, this is the
     // only pair of APIs in the library that have the same name for its results interface.
-    interface Result : List<Contact>, Redactable {
+    interface Result : List<Contact>, CrudApi.Result {
 
         // We have to cast the return type because we are not using recursive generic types.
         override fun redactedCopy(): Result
     }
 }
 
 @Suppress("FunctionName")
-internal fun BroadQuery(contacts: Contacts): BroadQuery = BroadQueryImpl(
-    contacts.applicationContext.contentResolver,
-    contacts.permissions,
-    contacts.customDataRegistry
-)
+internal fun BroadQuery(contacts: Contacts): BroadQuery = BroadQueryImpl(contacts)
 
 private class BroadQueryImpl(
-    private val contentResolver: ContentResolver,
-    private val permissions: ContactsPermissions,
-    private val customDataRegistry: CustomDataRegistry,
+    override val contactsApi: Contacts,
```

Comment:
Instead of keeping references to the individual properties that are already kept by the [Contacts] instance, just keep a reference to the [Contacts] instance 😁 

Actually, this pattern was already being used by several CRUD APIs prior to recent PRs.